lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -1383,6 +1418,27 @@ sub cancel{
     return __default(@_);
 }
 
+=item force
+
+Force work.  Forgets unmatching tests, applies a curing statement and move on.
+
+=cut
+
+sub force{
+    my ($request) = @_;
+    my $database = _init_db($request);
+
+    my %test = map { $_->name => $_ } LedgerSMB::Upgrade_Tests->get_tests();
+    my $force_queries = $test{$request->{check}}->{force_queries};
+
+    for my $force_query ( @$force_queries ) {
+        my $dbh = $request->{dbh};
+        $dbh->do($force_query);
+        $dbh->commit;
+    }
+    return upgrade($request);
```

Comment:
Hmm. But what if the strategy is "ok, thanks for notifying me, please proceed, I'm accepting the data-inconsistencies for now (Force)"? Doesn't re-running the upgrade without further measures then simply provide me the same result? Shouldn't we have an indication of upgrade checks that were forced and thus shouldn't be checked again?